awscrt/http.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import _aws_crt_python
+from concurrent.futures import Future
+from awscrt.io import ClientBootstrap, TlsConnectionOptions, SocketOptions
+
+
+# Represents an Http connection to a remote endpoint. Everything in this class is non-blocking.
+class HttpClientConnection(object):
+    __slots__ = ('_bootstrap', '_tls_connection_options', '_on_connection_shutdown', '_native_handle')
+
+    # don't call me, I'm private
+    def __init__(self, bootstrap, on_connection_shutdown, tls_connection_options):
```

Comment:
While there isn't full enforcement, you could do something like: https://stackoverflow.com/a/46459300 to be a bit more draconian

Reply:
I asked the python team which they hated least and this was their advisement.